components/FileSelector.vue

```diff
@@ -18,41 +12,46 @@
 
   const props = defineProps({
     multiple: { type: Boolean, required: true },
-    label: { type: String, required: true },
-    variable_to_update: { type: String, required: true },
-    variable_to_increment: { type: String, required: true },
+    key: { type: String, required: false, default: "" },
     schema: { type: Object, required: true },
   })
-  const { multiple, label, variable_to_update, variable_to_increment, schema } =
-    props
+
+  const { multiple, key, schema } = props
 
   const accept = ref("")
-  const files = ref([])
+  const loading = ref(false)
 
-  watch(files, (value) => {
-    stepper_tree[variable_to_update] = value
-    stepper_tree[variable_to_increment]++
-  })
+  const toggle_loading = useToggle(loading)
 
-  function fill_extensions(response) {
-    const extensions = response._data.extensions
-      .map((extension) => "." + extension)
-      .join(",")
-    accept.value = extensions
+  function files_uploaded_event(value) {
+    stepper_tree["files"] = value
+    if (value.length) {
+      stepper_tree["current_step_index"]++
+    }
   }
 
   async function get_allowed_files() {
+    toggle_loading()
+    const params = { key }
     await api_fetch(
-      { schema },
+      { schema, params },
       {
         response_function: (response) => {
-          fill_extensions(response)
+          accept.value = response._data.extensions
+            .map((extension) => "." + extension)
+            .join(",")
         },
       },
     )
+    toggle_loading()
   }
-
-  onMounted(async () => {
-    await get_allowed_files()
+  onMounted(() => {
+    get_allowed_files()
   })
 </script>
+
+<style scoped>
+  .v-btn {
+    text-transform: unset !important;
```

Comment:
next time, take a snapshot of the button with and without this and ask the team to vote